go/src/oauth2client/token.go

```diff
@@ -16,6 +20,14 @@ type Token struct {
 	// if it expires.
 	RefreshToken string `json:"refresh_token,omitempty"`
 
+	// Expiry is the optional expiration time of the access token.
+	//
+	// If zero, TokenSource implementations will reuse the same
+	// token forever and RefreshToken or equivalent
+	// mechanisms for that TokenSource will not be used.
+	Expiry time.Time `json:"expiry,omitempty"`
+	// contains filtered or unexported fields
+
 	// ExpiresIn is the optional expiration time in seconds.
-	ExpiresIn int `json:"expires_in,omitempty"`
+	ExpiresIn *int `json:"expires_in,omitempty"`
```

Comment:
"expires_in" is part of the oauth 2 spec, but I didn't see "expiry", which is defined in golang's token